src/convertToObject.js

```diff
@@ -1,12 +1,68 @@
 'use strict';
 
-/**
- * @param {string} sourceString
- *
- * @return {object}
- */
 function convertToObject(sourceString) {
-  // write your code here
+  if (sourceString === '') {
+    return {};
+  }
+
+  const firstObrabotka = sourceString;
+
+  const bigArr = firstObrabotka.split(';');
+
+  const arrFinal = bigArr.filter((str) => str.trim() !== '');
+
+  let smallArr;
+  const result = {};
```

Comment:
The variable 'smallArr' is not needed outside the loop. Declare it inside the loop to limit its scope and improve readability.